File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int can = nums[0];
        int count = 1;
        for(int i = 1; i < nums.length;++i){
            if(count == 0){
                can = nums[i];
                count = 1;
            }
            else if(can == nums[i]){
                count += 1;
            }else{
                count -= 1;
            }
        }
        return can;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        can = nums[0]
        for i in range(1,len(nums)):
            if count == 0:
                can = nums[i]
                count = 1
            elif can == nums[i]:
                count += 1
            else:
                count -= 1
        return can
